src/sinks/datadog/logs/config.rs

```diff
@@ -31,7 +30,6 @@ pub struct DatadogLogsConfig {
     // Deprecated name
     #[serde(alias = "api_key")]
     default_api_key: String,
-    pub(crate) encoding: EncodingConfig<Encoding>,
```

Comment:
I think we want to just strike `encoding.codec`. The other encoding fields like `only_fields` and `except_fields` could still be useful to users. You can see the `new_relic_logs` sink for an example of this:

https://github.com/timberio/vector/blob/4fecc5fdd7528ce25811fa4871c4fc65742250a2/src/sinks/new_relic_logs.rs#L52

Reply:
Addressed in 0eabbf1